### PR TITLE
Initial removal of IcePatch2 from IceGrid

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR is the first step towards removing IcePatch2 (and server / application distribs) from IceGrid.

This feature was deprecated in Ice 3.7.

### CODE REVIEW

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Application/PlainServer.java [hunk 1]
```diff
@@ -73,7 +68,7 @@ static public ServerDescriptor newServerDescriptor()
             "",
             "",
             true,
-            new DistributionDescriptor("", new java.util.LinkedList<String>()),
+            new DistributionDescriptor("", new java.util.LinkedList<String>()), // dummy
```
Comment (pepone) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Application/PlainServer.java:71]: We have to keep this for backward compatibility, right?

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/ServerEditor.java [hunk 2]
```diff
@@ -56,11 +56,6 @@ public JToolBar getToolBar()
         _options.setEditable(false);
         _user.setEditable(false);
 
-        _allocatable.setEnabled(false);
```
Comment (pepone) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/ServerEditor.java:59]: Why did you change `_allocatable` here?

**Review by pepone — APPROVED** [review]

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/ServerEditor.java (hunk 2 above)
Comment (bernardnormier) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/ServerEditor.java:59]: This was an accidental change. Will fix. Good catch!

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Application/PlainServer.java (hunk 1 above)
Comment (bernardnormier) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Application/PlainServer.java:71]: Yes. It's not so much for the IceGridGUI - IceGrid communications but for the database format.

**Comment by InsertCreativityHere** [comment]
> Ubuntu build failing in ...

These are the same issues I ran into with the Slice compiler.
I'll merge in my PR for updating the Slice compilers, and then I'll port those changes over to the other 2 compilers.

**Review by InsertCreativityHere — APPROVED** [review]
Looks good!
